### PR TITLE
Allow plugin names to be up to 60 characters instead of 40

### LINKED ISSUE #18727: Restrict plugin name to 60 characters
Follow up from https://github.com/matomo-org/matomo/pull/18723/files#r797936437

We have been using 40 characters in some columns, in some others 60 for `plugin_name`.

We should adjust `isValidPluginName` method to check the length in the regex and if not valid, throw an exception? It may be painful for some people to rename the plugin but at least on Marketplace the length has been already enforced.

The Marketplace itself enforces a length of 40 characters. Happy to increase this to 60 characters if better. 

We want to make sure that

* Marketplace and Matomo itself validate plugin name
* All DB columns use same length

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

Fixes: #18727
adding plugin name checks with 60 characters limit

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: tests/PHPUnit/Integration/Plugin/ManagerTest.php [hunk 1]
```diff
@@ -185,9 +185,19 @@ public function getPluginNameProvider()
             array(false, 'a-ererer'),
             array(false, '..'),
             array(false, '/'),
+            array(true, 'a'.$this->generateRandomString(59)),
+            array(false, 'a'.$this->generateRandomString(60)),
         );
     }
-
+   private function generateRandomString($length = 10) {
```
Comment (sgiehl) [inline tests/PHPUnit/Integration/Plugin/ManagerTest.php]: We already have a method that does mostly the same: `Common::getRandomString()`

**Review by sgiehl — CHANGES_REQUESTED** [review]
@peterhashair Did you check the other code for required changes? There is at least
https://github.com/matomo-org/matomo/blob/115527353a9e75e01aa4d263408956ae45403bea/plugins/CoreConsole/Commands/GeneratePlugin.php#L137-L139
which needs to be removed. Not sure if there are other places where similar stuff is checked.

File: tests/PHPUnit/Integration/Plugin/ManagerTest.php (hunk 1 above)
Comment (peterhashair) [reply tests/PHPUnit/Integration/Plugin/ManagerTest.php]: @sgiehl ops, my bad, update it. :)

**Comment by peterhashair** [comment]
@sgiehl I couldn't find anywhere else that has a 40 char limit, anywhere I need to be aware?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ The Product Changelog at **[matomo.org/changelog](https://matomo.org/changelog)*
 
 ### Breaking Changes
 
+* Limited plugin name to 60 characters.
```
Comment (sgiehl) [inline CHANGELOG.md]: Might be better to write something more descriptive and helpful:
```suggestion
* Plugin names are now limited to 60 characters. If you used to have a plugin with a longer name, you might need to rename it.
```

File: tests/PHPUnit/Integration/Plugin/ManagerTest.php [hunk 3]
```diff
@@ -174,7 +174,7 @@ public function getPluginNameProvider()
             array(true, 'a0'),
             array(true, 'pluginNameTest'),
             array(true, 'PluginNameTest'),
-            array(true, 'PluginNameTest92323232eerwrwere938'),
+            array(true, 'PluginNameTest'.$this->generateRandomString()),
```
Comment (sgiehl) [inline tests/PHPUnit/Integration/Plugin/ManagerTest.php]: Guess this needs to be changed as well, after removing the method...

**Review by sgiehl — CHANGES_REQUESTED** [review]

**Comment by peterhashair** [comment]
This PR maybe require other docs to update.

**Review by sgiehl — APPROVED** [review]

**Pull request merged**